tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,198 @@
+
+		BeforeEach(func() {
+			By("Create NetworkAttachmentDefinition")
+			Expect(createNAD(virtClient, util.NamespaceTestDefault, nadName)).To(Succeed())
+
+			defaultBridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding(primaryNetwork)
+			secondaryLinuxBridgeInterface1 := libvmi.InterfaceDeviceWithBridgeBinding(secondaryNetwork1.Name)
+			secondaryLinuxBridgeInterface2 := libvmi.InterfaceDeviceWithBridgeBinding(secondaryNetwork2.Name)
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&defaultBridgeInterface, primaryInterfaceMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&secondaryLinuxBridgeInterface1, secondaryInterface1Mac)),
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&secondaryLinuxBridgeInterface2, secondaryInterface2Mac)),
+				libvmi.WithNetwork(secondaryNetwork1),
+				libvmi.WithNetwork(secondaryNetwork2),
+				libvmi.WithCloudInitNoCloudUserData(manipulateGuestLinksScript(primaryInterfaceNewMac, dummyInterfaceMac), false))
+
+			var err error
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        primaryInterfaceMac,
+					Name:       primaryNetwork,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceDomainAndGA,
+					InterfaceName: "eth1",
+					MAC:           secondaryInterface1Mac,
+					Name:          secondaryInterface1Name,
+				},
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        secondaryInterface2Mac,
+					Name:       secondaryInterface2Name,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "eth0",
+					MAC:           primaryInterfaceNewMac,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: dummyInterfaceName,
+					MAC:           dummyInterfaceMac,
+				},
+			}
+
+			var vmiObj *kvirtv1.VirtualMachineInstance
+			Eventually(func() bool {
+				var err error
+				vmiObj, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				return dummyInterfaceExists(vmiObj)
+			}, 120*time.Second, 2*time.Second).Should(Equal(true))
+
+			Expect(len(vmiObj.Status.Interfaces)).To(Equal(len(expectedInterfaces)),
+				fmt.Sprintf("interfaces doesn't match\n%v\n%v", expectedInterfaces, vmiObj.Status.Interfaces))
+
+			orderedInterfaces := orderInterfaceStatus(vmiObj.Status.Interfaces)
+			for i := range orderedInterfaces {
+				expectedInterfaces[i].IP = orderedInterfaces[i].IP
+				expectedInterfaces[i].IPs = orderedInterfaces[i].IPs
```

Comment:
I do not understand this one. You are overriding the expected interfaces with the reported ones IPs? Is this in order to normalize them for comparison?
If this is the case, it is confusing, because you involve both objects. Can you just remove the IP data from the current state and that is it?

Reply:
for some reason it didnt work in the past, but now it does, done